internal/service/teamprojectassignment/resource_schema.go

```diff
@@ -26,6 +28,19 @@ func resourceSchema() schema.Schema {
 	}
 }
 
+func dataSourceSchema() dsschema.Schema {
+	return conversion.DataSourceSchemaFromResource(resourceSchema(), &conversion.DataSourceSchemaRequest{
+		RequiredFields: []string{"project_id", "team_id"},
+		OverridenFields: map[string]dsschema.Attribute{
+			"role_names": dsschema.SetAttribute{
+				Computed:            true,
+				ElementType:         types.StringType,
+				MarkdownDescription: "One or more project-level roles to assign to the team.",
+			},
```

Comment:
q: Why do we need the override of this attribute? Would expect the default behaviour of DataSourceSchemaFromResource be to make `role_names` computed